### PR TITLE
Cast string values to float

### PR DESCRIPTION
**Pull request by EZarkov**
Cast string values to integer in constructor and on unserialize.
It is for PHP 7.1 compatibility:
http://php.net/manual/en/migration71.other-changes.php#migration71.other-changes.apprise-on-arithmetic-with-invalid-strings

### CODE REVIEW

**Comment by hkdobrev** [comment]
@EZarkov Could you please add a failing test which would fail without these changes and pass with them? Thanks!

File: classes/Kohana/Jam/Range.php [hunk 1]
```diff
@@ -59,13 +59,13 @@ public function __construct($source = NULL, $format = NULL)
...
 		{
-			$source = explode('|', $source);
+			$source = array_map('intval', explode('|', $source));
 		}
 
 		if (is_array($source))
 		{
-			$this->min($source[0]);
-			$this->max($source[1]);
+			$this->min((int) $source[0]);
```
Comment (phgeorgiev) [inline classes/Kohana/Jam/Range.php]: @EZarkov You can add the `array_map` call after converting the string to an array, this way will be more consistent.

File: classes/Kohana/Jam/Range.php [hunk 2]
```diff
@@ -169,7 +169,7 @@ public function unserialize($data)
 	{
 		list($min, $max) = explode('|', $data);
 
-		$this->min($min);
-		$this->max($max);
+		$this->min((int) $min);
```
Comment (phgeorgiev) [inline classes/Kohana/Jam/Range.php]: Same here

File: classes/Kohana/Jam/Range.php (hunk 1 above)
Comment (EZarkov) [inline classes/Kohana/Jam/Range.php]: @phgeorgiev 
Why i have to walk through array and add additional complexity when there is easier way to cast parameter  on assigning?

File: classes/Kohana/Jam/Range.php (hunk 2 above)
Comment (EZarkov) [inline classes/Kohana/Jam/Range.php]: Same here

**Pull request merged**